gointelowl/job.go

```diff
@@ -0,0 +1,242 @@
+	requestUrl := fmt.Sprintf("%s/api/jobs/%d/download_sample", jobService.client.options.Url, jobId)
+	request, err := http.NewRequest("GET", requestUrl, nil)
+	if err != nil {
+		return nil, err
+	}
+	successResp, err := jobService.client.newRequest(ctx, request)
+	if err != nil {
+		return nil, err
+	}
+	return successResp.Data, nil
+}
+
+/*
+* Desc: Delete a Job through its ID
+* Endpoint: DELETE /api/jobs/{jobID}
+ */
+func (jobService *JobService) Delete(ctx context.Context, jobId uint64) (bool, error) {
+	requestUrl := fmt.Sprintf("%s/api/jobs/%d", jobService.client.options.Url, jobId)
+	request, err := http.NewRequest("DELETE", requestUrl, nil)
+	if err != nil {
+		return false, err
+	}
+	successResp, err := jobService.client.newRequest(ctx, request)
+	if err != nil {
+		return false, err
+	}
+	if successResp.StatusCode == http.StatusNoContent {
+		return true, nil
+	}
+	return false, nil
+}
+
+/*
+* Desc: Stop a running job through its ID
+* Endpoint: PATCH /api/jobs/{jobID}/kill
+ */
+func (jobService *JobService) Kill(ctx context.Context, jobId uint64) (bool, error) {
+	requestUrl := fmt.Sprintf("%s/api/jobs/%d/kill", jobService.client.options.Url, jobId)
+	request, err := http.NewRequest("PATCH", requestUrl, nil)
+	if err != nil {
+		return false, err
+	}
+	successResp, err := jobService.client.newRequest(ctx, request)
+	if err != nil {
+		return false, err
+	}
+	if successResp.StatusCode == http.StatusNoContent {
+		return true, nil
+	}
+	return false, nil
+}
+
+/*
+* Desc: Stop a running analyzer on a job that is being processed through its ID and the analyzer's name
+* Endpoint: PATCH /api/jobs/{jobID}/analyzer/{nameOfAnalyzer}/kill
+ */
+func (jobService *JobService) KillAnalyzer(ctx context.Context, jobId uint64, analyzerName string) (bool, error) {
+	requestUrl := fmt.Sprintf("%s/api/jobs/%d/analyzer/%s/kill", jobService.client.options.Url, jobId, analyzerName)
+	request, err := http.NewRequest("PATCH", requestUrl, nil)
+	if err != nil {
+		return false, err
+	}
+	successResp, err := jobService.client.newRequest(ctx, request)
+	if err != nil {
+		return false, err
+	}
+	if successResp.StatusCode == http.StatusNoContent {
+		return true, nil
+	}
+	return false, nil
+}
+
+/*
+* Desc: Re-run a selected analyzer on a job that is being processed through its ID and the analyzer's name
+* Endpoint: PATCH /api/jobs/{jobID}/analyzer/{nameOfAnalyzer}/retry
+ */
+func (jobService *JobService) RetryAnalyzer(ctx context.Context, jobId uint64, analyzerName string) (bool, error) {
+	requestUrl := fmt.Sprintf("%s/api/jobs/%d/analyzer/%s/retry", jobService.client.options.Url, jobId, analyzerName)
+	request, err := http.NewRequest("PATCH", requestUrl, nil)
+	if err != nil {
+		return false, err
+	}
+	successResp, err := jobService.client.newRequest(ctx, request)
+	if err != nil {
+		return false, err
+	}
+	if successResp.StatusCode == http.StatusNoContent {
```

Comment:
Since this is a http client, we should be returning the response from server and not parsing the result.

Reply:
So the return type should be `http.Response` and not `bool`

Reply:
It makes sense to abstract the response to a boolean